### PR TITLE
Add a fallback phylo_dist_path

### PR DESCRIPTION
**Pull request by gonzalobenegas**
- Adds a new `load_model` function for GPN-Star that tries `model_path/phylo_dist` as falback `phylo_dist_path`. This simplified loading of models on HuggingFace.
- Modifies `logits.py` to use the new function (could be added to other places as well).

### CODE REVIEW

**Comment by gonzalobenegas** [comment]
@cz-ye what do you think?
